### PR TITLE
JBPM-6788: Stunner - Add BPMN Inclusive Gateway

### PR DESCRIPTION
**Pull request by wmedvede**

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @romartin @hasys @jsoltes would you mind review this PR ?
Thanks.

**Comment by romartin** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 1]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                .addTrueCase(new OnEntryAction(),
+                             new OnEntryAction())
 
-                        .addTrueCase(new OnEntryAction(new ScriptTypeListValue()),
-                                     new OnEntryAction(new ScriptTypeListValue()))
-                        .build()
-        );
+                .addTrueCase(new OnEntryAction(new ScriptTypeListValue()),
+                             new OnEntryAction(new ScriptTypeListValue()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 2]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                                                              new OnEntryAction(),
+                                                              new OnExitAction(),
+                                                              new IsAsync(),
+                                                              new AdHocAutostart()),
+                             new BusinessRuleTaskExecutionSet(new RuleFlowGroup(),
+                                                              new OnEntryAction(),
+                                                              new OnExitAction(),
+                                                              new IsAsync(),
+                                                              new AdHocAutostart()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 3]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
-        );
+                .addTrueCase(new AdHocSubprocessTaskExecutionSet(new AdHocCompletionCondition(),
+                                                                 new AdHocOrdering(),
+                                                                 new OnEntryAction(),
+                                                                 new OnExitAction()),
+                             new AdHocSubprocessTaskExecutionSet(new AdHocCompletionCondition(),
+                                                                 new AdHocOrdering(),
+                                                                 new OnEntryAction(),
+                                                                 new OnExitAction()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 4]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                .addTrueCase(new OnExitAction(),
+                             new OnExitAction())
 
-                        .addTrueCase(new OnExitAction(new ScriptTypeListValue()),
-                                     new OnExitAction(new ScriptTypeListValue()))
-                        .build()
-        );
+                .addTrueCase(new OnExitAction(new ScriptTypeListValue()),
+                             new OnExitAction(new ScriptTypeListValue()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 5]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                .addTrueCase(new Script(),
+                             new Script())
 
-                        .addTrueCase(new Script(new ScriptTypeValue()),
-                                     new Script(new ScriptTypeValue()))
-                        .build()
-        );
+                .addTrueCase(new Script(new ScriptTypeValue()),
+                             new Script(new ScriptTypeValue()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 6]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                .addTrueCase(new ScriptTaskExecutionSet(new Script(),
+                                                        new IsAsync()),
 
-                                     new ScriptTaskExecutionSet(new Script(),
-                                                                new IsAsync()))
-                        .build()
-        );
+                             new ScriptTaskExecutionSet(new Script(),
+                                                        new IsAsync()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 7]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                                                      new IsAsync(),
+                                                      new Skippable(),
+                                                      new Priority(),
+                                                      new Subject(),
+                                                      new Description(),
+                                                      new CreatedBy(),
+                                                      new AdHocAutostart(),
+                                                      new OnEntryAction(),
+                                                      new OnExitAction()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 8]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                                                  new BackgroundSet(),
+                                                  new FontSet(),
+                                                  new CircleDimensionSet(),
+                                                  new GatewayExecutionSet()),
+                             new InclusiveGateway(new BPMNGeneralSet(),
+                                                  new BackgroundSet(),
+                                                  new FontSet(),
+                                                  new CircleDimensionSet(),
+                                                  new GatewayExecutionSet()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 9]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
-                                     new ScriptTypeListValue().addValue(new ScriptTypeValue(null,
-                                                                                            "b")))
+    @Test
+    public void testGatewayExecutionSetEqualsAndHashCode() {
+        TestCaseBuilder.newTestCase()
+                .addTrueCase(new GatewayExecutionSet(),
+                             new GatewayExecutionSet())
+                .addTrueCase(new GatewayExecutionSet(new DefaultRoute()),
+                             new GatewayExecutionSet(new DefaultRoute()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 10]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
-                                      new ScriptTypeListValue().addValue(new ScriptTypeValue("X",
-                                                                                             "b")))
+    @Test
+    public void testDefaultRouteEqualsAndHashCode() {
+        TestCaseBuilder.newTestCase()
+                .addTrueCase(new DefaultRoute(),
+                             new DefaultRoute())
+                .addTrueCase(new DefaultRoute("A"),
+                             new DefaultRoute("A"))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 11]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
-                                      new ScriptTypeListValue().addValue(new ScriptTypeValue("a",
-                                                                                             "Y")))
+    @Test
+    public void testCircleDimensionSetEqualsAndHashCode() {
+        TestCaseBuilder.newTestCase()
+                .addTrueCase(new CircleDimensionSet(),
+                             new CircleDimensionSet())
+                .addTrueCase(new CircleDimensionSet(new Radius()),
+                             new CircleDimensionSet(new Radius()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 12]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+    @Test
+    public void testBPMNGeneralSetEqualsAndHashCode() {
+        TestCaseBuilder.newTestCase()
+                .addTrueCase(new BPMNGeneralSet(),
+                             new BPMNGeneralSet())
+                .addTrueCase(new BPMNGeneralSet(new Name(),
+                                                new Documentation()),
+                             new BPMNGeneralSet(new Name(),
+                                                new Documentation()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 13]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+        TestCaseBuilder.newTestCase()
+                .addTrueCase(new BackgroundSet(),
+                             new BackgroundSet())
+                .addTrueCase(new BackgroundSet(new BgColor(),
+                                               new BorderColor(),
+                                               new BorderSize()),
+                             new BackgroundSet(new BgColor(),
+                                               new BorderColor(),
+                                               new BorderSize()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java [hunk 14]
```diff
@@ -943,382 +960,445 @@ public void testEmbeddedSubprocessHashCode() {
...
+                                         new FontColor(),
+                                         new FontSize(),
+                                         new FontBorderSize(),
+                                         new FontBorderColor()),
+                             new FontSet(new FontFamily(),
+                                         new FontColor(),
+                                         new FontSize(),
+                                         new FontBorderSize(),
+                                         new FontBorderColor()))
+                .test();
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/test/java/org/kie/workbench/common/stunner/bpmn/definition/HashCodeAndEqualityTest.java]: Only True Cases?

**Pull request merged**